docs: add types documentation (#335)
litestar-org/advanced-alchemy: docs/usage/types.rst

    
    class MyModel:
        id = Column(BigIntIdentity, primary_key=True)

JsonB
-----

A JSON type that uses the most efficient JSON storage for each database:

- PostgreSQL/CockroachDB: Uses native JSONB
- Oracle: Uses Binary JSON (BLOB with JSON constraint)
- Others: Uses standard JSON type

.. code-block:: python

    from advanced_alchemy.types import JsonB
    
    class MyModel:
        data = Column(JsonB)

Type Features
-------------

All types include:

- Proper Python type annotations for modern IDE support
- Automatic dialect-specific implementations
- Consistent behavior across different database backends
- Integration with SQLAlchemy's type system

Usage Example
-------------

Here's a complete example using multiple types:

.. code-block:: python

    from sqlalchemy import Column
    from advanced_alchemy.types import (
        DateTimeUTC,
        EncryptedString,
        GUID,
        JsonB,
    )

    class User:
        id = Column(GUID, primary_key=True)
        created_at = Column(DateTimeUTC)
        password = Column(EncryptedString(key="secret-key"))
        preferences = Column(JsonB)

Using Types with Alembic
------------------------

> When using these types with Alembic migrations, you need to properly configure your ``script.py.mako`` template. The key is to make the custom types available through the ``sa`` namespace that Alembic uses.

Reviewer: ```suggestion
If you are not using Advanced Alchemy's built-in `alembic` templates, you need to properly configure your ``script.py.mako`` template. The key is to make the custom types available through the ``sa`` namespace that Alembic uses.
```
